### PR TITLE
enable pmw3901 flow if (new) parameter SENS_EN_PMW_FLOW is true

### PR DESCRIPTION
**Pull request by BazookaJoe1900**
**Describe problem solved by the proposed pull request**
the startup scripts try to start the pmw3901 flow on some boards by default. even if the drone doesn't has one connected.
that cause false errors on startup script for drones that doesn't has the sensors (most of the configurations)

> ERROR [pmw3901] driver start failed

**Describe your preferred solution**
New SENS_EN_PMW_FLOW parameter that allows to define if the sensor should be or not on the system.

**Describe possible alternatives**
With this parameter, starting the flow can moved to common rc.sensors file. curently its only on boards that compile the pmw3901.
In that case, boards that don't have the driver compiled will show "Parameter SENS_EN_PMW_FLOW not found" (that is being delt on #12832)
does that prefered?

### CODE REVIEW

**Comment by BazookaJoe1900** [comment]
Another thing, One AV/ax-1 I didn't add the code, because there is strange 'if' that if there is no flow there is adis16497. Who can address what to do there? I don't know the reason for current implementation

**Comment by dagar** [comment]
How about `SENS_EN_PMW3901` and move it more centrally? Boards that don't support it won't even have the parameter (it would be pruned).

**Comment by BazookaJoe1900** [comment]
@dagar done.
small quistion, the compretion tell me there is no EOL in the end of some scripts. Just want to make sure that its ok.
image

**Comment by LorenzMeier** [comment]
@BazookaJoe1900 No, that's not ok - please add the newlines.

**Comment by BazookaJoe1900** [comment]
Ok, I will fix that later.
What about ax1?  Currently there is double call to start the driver.

**Comment by dagar** [comment]
> Ok, I will fix that later.
> What about ax1? Currently there is double call to start the driver.

For av-x check the parameter, but only to start the ADIS16977 if the PMW3901 is disabled.

File: boards/px4/fmu-v5/init/rc.board_sensors [hunk 1]
```diff
@@ -24,9 +24,3 @@ lis3mdl -X start
 
 # Possible internal compass
 ist8310 -C -b 5 start
-
-# Possible pmw3901 optical flow sensor
-pmw3901 start
-
-# Internal SPI
-ms5611 -s start
```
Comment (sfuhrer) [inline boards/px4/fmu-v5/init/rc.board_sensors]: Why disable the baro?

Fixed with https://github.com/PX4/Firmware/pull/12882

**Pull request merged**